Mathlib/Geometry/Manifold/Orientable.lean

```diff
@@ -0,0 +1,135 @@
+/-
+Copyright (c) 2024 Rida Hamadani. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Rida Hamadani
+-/
+import Mathlib.Geometry.Manifold.SmoothManifoldWithCorners
+
+/-!
+# Orientable Manifolds
+
+This module defines orientable manifolds.
+
+## Main Definitions
+
+- `OrientationPreserving` : a map is orientation-preserving on a given set if the determinant of
+  its Jacobian is strictly positive on that set.
+- `orientationPreservingGroupoid` : the groupoid of partial homeos of `H` which are
+  orientation-preserving.
+- `OrientableManifold`: a type class saying that the charted space `M`, modelled on the space `H`,
+  admis an orientation.
+
+## TODO
+
+- Define orientable `0`-manifolds.
```

Comment:
IIUC, this PR already handles the 0-dimensional case (and correctly). Instead, I would extend this TODO slightly differently: I'll leave this blank for now and try to return here after dinner.

Reply:
So, here's an initial list of what I'd write.
- try to generalise this discussion to any non-trivially normed field
- on a given connected set, a diffeomorphism is either orientation-preserving or orientation-reversing
- a normed space (with the trivial model) is orientable
- the n-sphere is orientable
- products of orientable manifolds are orientable

- define orientations of a smooth manifold, and show that a manifold is orientable if and only if it admits an orientation
- define orientation-preserving and reserving maps between manifolds